org.mixedrealitytoolkit.input/Interactors/GazePinch/GazePinchInteractor.cs

```diff
@@ -444,5 +452,39 @@ protected override void OnSelectExited(SelectExitEventArgs args)
         }
 
         #endregion XRBaseInteractor
+
+        /// <summary>
+        /// Updates the pinch state of the GazePinchInteractor.
+        /// If handedness is not set then it defaults to right hand.
+        /// If the pinch data is not available for the set hand then the other hand is tried.
+        /// </summary>
+        private void UpdatePinchState()
+        {
+            if (logicalSelectState == null)
+            {
+                Debug.LogWarning("GazePinchInteractor is missing logicalSelectState, pinch state won't update.");
+                return;
+            }
+
+            if (XRSubsystemHelpers.HandsAggregator == null)
+            {
+                Debug.LogWarning("XRSubsystemHelpers.HandsAggregator is null, pinch state won't update.");
+                return;
+            }
+
+            var xrNode = handedness.ToXRNode();
+            bool gotPinchData = XRSubsystemHelpers.HandsAggregator.TryGetPinchProgress(xrNode,
+                out bool isPinchReady, out bool isPinching, out float pinchAmount);
+            if (!gotPinchData) //Try the other hand if the set hand does not have pinch data.
```

Comment:
@ms-RistoRK I'm a little confused why we try the other hand. Aren't `GazePinchInteractor`s explicitly handed? So there should be a second one in the scene that will handle the other hand.